test/e2e/client_tls_test.go

```diff
@@ -1222,15 +1191,33 @@ func TestCRLUpdate(t *testing.T) {
 			},
 		},
 	}
-	for _, tc := range testCases {
+	namespaceName := names.SimpleNameGenerator.GenerateName("crl-update-")
+	namespace := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if err := kclient.Create(context.TODO(), &namespace); err != nil {
+		t.Fatalf("Failed to create namespace %q: %v", namespace.Name, err)
+	}
+	t.Cleanup(func() { assertDeleted(t, kclient, &namespace) })
+	for i := range testCases {
+		tc := testCases[i]
```

Comment:
What's the point of this change?  Did you mean to use `tc := &testCases[i]` to avoid copying the whole test case?  Even that would be an unnecessary micro-optimization test code, in my opinion, but I don't understand the purpose of `tc := testCases[i]` at all.

Reply:
`for _, tc := range testCases` isn't thread safe. I initially had it that way, but once `t.Parallel()` was called, `tc` in all threads converged on the last test case in the list.